cfgmgr/intfmgr.cpp

```diff
@@ -576,10 +529,57 @@ bool IntfMgr::doIntfGeneralTask(const vector<string>& keys,
             }
 
             // set STATE_DB port state
-            setSubIntfStateOk(subIntfAlias);
+            setSubIntfStateOk(alias);
         }
-        m_appIntfTableProducer.set(subIntfAlias.empty() ? alias : subIntfAlias, data);
-        m_stateIntfTable.hset(subIntfAlias.empty() ? alias : subIntfAlias, "vrf", vrf_name);
+
+        if (!vrf_name.empty())
+        {
+            setIntfVrf(alias, vrf_name);
+        }
+
+        /*Set the mac of interface*/
+        if (!mac.empty())
+        {
+            setIntfMac(alias, mac);
+        }
+        else
+        {
+            FieldValueTuple fvTuple("mac_addr", MacAddress().to_string());
+            data.push_back(fvTuple);
+        }
+
+        if (!proxy_arp.empty())
+        {
+            if (!setIntfProxyArp(alias, proxy_arp))
+            {
+                SWSS_LOG_ERROR("Failed to set proxy ARP to \"%s\" state for the \"%s\" interface", proxy_arp.c_str(), alias.c_str());
+                return false;
+            }
+
+            if (!alias.compare(0, strlen(VLAN_PREFIX), VLAN_PREFIX))
+            {
+                FieldValueTuple fvTuple("proxy_arp", proxy_arp);
+                data.push_back(fvTuple);
+            }
+        }
+
+        if (!grat_arp.empty())
+        {
+            if (!setIntfGratArp(alias, grat_arp))
+            {
+                SWSS_LOG_ERROR("Failed to set ARP accept to \"%s\" state for the \"%s\" interface", grat_arp.c_str(), alias.c_str());
+                return false;
+            }
+
+            if (!alias.compare(0, strlen(VLAN_PREFIX), VLAN_PREFIX))
+            {
+                FieldValueTuple fvTuple("grat_arp", grat_arp);
+                data.push_back(fvTuple);
+            }
+        }
+
+        m_appIntfTableProducer.set(alias, data);
```

Comment:
Does orchagent requires similar change to handle sub-interface RIF configurations?

Reply:
No, ASIC_DB output looks good.